drop-transfer/src/ws/server/mod.rs

```diff
@@ -763,7 +765,36 @@ impl FileXferTask {
                 return Err(crate::Error::UnexpectedData);
             }
 
-            downloader.validate(tmp_loc).await?;
+            if emit_checksum_events {
+                events.checksum_start().await;
+
+                let progress_chan = tokio::sync::watch::channel(0u64).0;
+
+                tokio::spawn({
```

Comment:
Why separate tasks for events? The flow is like this:
`-> oneshot channel send -> oneshot channel recv -> event channel send ->`
I'd say it can be shortened to
`-> event channel sed ->`

Reply:
I wanted to add some throttling in between so it wouldn't be emitted for every 8K increment, however in hindsight I could have just increased the checksumming buffer.

I am fine with both ways personally, this one I prefer because it also allows for isolated channel just for the progress instead of allowing sending any event by passing the regular channel

Reply:
Last commit contains the rework, now it's callback based and the throttling is done by increasing checksum size to 256kb